### PR TITLE
Schemafy Binary Feature

### PR DESCRIPTION
**Pull request by arnavgarg1**
Would like a careful review on this PR, specifically the output feature and decoders (loss looks fine). It would also be great if we can review the metrics (I updated it to reflect the current state of binary metrics today, but any confirmation that this looks right would be great).

I also added a mermaid graph for the regressor decoder (which is effectively the same as a classifier decoder but uses sigmoid activation)

### CODE REVIEW

File: docs/configuration/features/binary_features.md [hunk 1]
```diff
@@ -116,64 +88,69 @@ decoder:
...
+{{ render_fields(schema_class_to_fields(loss, exclude=["type"]), details=details) }}
+
+Loss and loss related parameters can also be defined once and applied to all binary output features using the [Type-Global Loss](../defaults.md#type-global-loss) section.
+
+### Metrics
+
+The metrics that are calculated every epoch and are available for binary features are the `accuracy`, `loss`,
+`precision`, `recall`, `roc_auc` and `specificity`.
+
+You can set either to be the `validation_metric` in the `training` section of the configuration if the `validation_field` is set as the name of a binary feature.
```
Comment (tgaddair) [inline docs/configuration/features/binary_features.md:156]: Replace: "You can set either" -> "You can set any of these"

File: docs/configuration/features/binary_features.md [hunk 2]
```diff
@@ -59,46 +36,41 @@ encoder:
...
+Binary output features can be used when a binary classification needs to be performed or when the output is a single probability. There are two decoders available: `regressor` and `passthrough`.
+
+Example binary output feature using default parameters:
+
 ```yaml
 name: binary_column_name
 type: binary
 reduce_input: sum
 dependencies: []
+calibration: false
```
Comment (tgaddair) [inline docs/configuration/features/binary_features.md:73]: Nice catch. I jut did a quick check of all the params for binary output feature as follows:

```
from ludwig.schema.features.binary_feature import BinaryOutputFeatureConfig
print(BinaryOutputFeatureConfig())
```

Looks like the one other param worth calling out is `threshold`, which is no longer part of the loss, as it was before. I would add its default `threshold: 0.5` in the YAML here and also add its param description to the list just below.

**Review by tgaddair — COMMENTED** [review]
Metrics look good, just need to add in `threshold`.

File: docs/configuration/features/binary_features.md [hunk 3]
```diff
@@ -116,64 +88,69 @@ decoder:
...
+The regressor decoder is a (potentially empty) stack of fully connected layers, followed by a projection into a single number followed by a sigmoid function.
+
+{% set decoder = get_decoder_schema("binary", "regressor") %}
+{{ render_yaml(decoder, parent="decoder") }}
+
+Parameters:
+
+{{ render_fields(schema_class_to_fields(decoder, exclude=["type"]), details=details) }}
+
+#### Passthrough Decoder
```
Comment (tgaddair) [inline docs/configuration/features/binary_features.md:129]: As discussed: let's remove this one.

File: docs/configuration/features/binary_features.md (hunk 2 above)
Comment (arnavgarg1) [reply docs/configuration/features/binary_features.md:73]: Done!

**Review by tgaddair — APPROVED** [review]

**Pull request merged**